src/odemis/driver/andorcam2.py

```diff
@@ -743,7 +743,7 @@ def __init__(self, name, role, device=None, emgains=None, shutter_times=None,
 
         # For temporary stopping the acquisition (kludge for the andorshrk
         # SR303i which cannot communicate during acquisition)
-        self.hw_lock = threading.Lock() # to be held during DRV_ACQUIRING (or shrk communicating)
+        self.hw_lock = threading.RLock()  # to be held during DRV_ACQUIRING (or shrk communicating)
```

Comment:
Maybe a comment on which methods would use the recursive functionality would be nice?
But that is not essential.